AWSConfig.txt
"""
Django settings for Backend project.

Generated by 'django-admin startproject' using Django 4.0.6.

For more information on this file, see
https://docs.djangoproject.com/en/4.0/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.0/ref/settings/
"""

from pathlib import Path
import os

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__))) 
REAL_BASE_DIR = Path(__file__).resolve().parent.parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.0/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-oe^*sm5lluuhb)gm8pcu$xa$!kdwu48j+#%1h86gt=d8()$k!k'
 

#ALLOWED_HOSTS = ['*']
ALLOWED_HOSTS = ['3.7.45.122','ec2-3-7-45-122.ap-south-1.compute.amazonaws.com']

 


# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles', 
    'rest_framework',#newly added https://www.bezkoder.com/django-react-axios-rest-framework/
    'corsheaders',#newly added https://www.bezkoder.com/react-crud-web-api/
    'MT_User', #newly added 
    'MT_Investor_Verify_Standards', 
    'MT_Investor_Verifiy_Nationality', 
    'MT_Investor_Kyc',
    'MT_Investor_Payment',
    'MT_Investor_Choose_Sector',
    'MT_Raise',
    'MT_Startup_Agreements',
    'MT_Startup_Companyinfo',
    'MT_Startup_Teaminfo',
    'MT_Startup_Pitchinfo',
    'MT_Startup_Pitch_Probleminfo',
    'MT_Startup_Pitch_Solution',
    'MT_Startup_Pitch_Productinfo',
    'MT_Startup_Pitch_Tractioninfo',
    'MT_Startup_Pitch_Businfo',
    'MT_Startup_Pitch_Competitioninfo',
    'MT_Startup_Pitch_Customerinfo',
    'MT_Startup_Pitch_Usagefundsinfo',
    'MT_Startup_Pitch_Visioninfo',
    'MT_Startup_Pitch_PotentialReturnsinfo',
    'MT_Startup_Banner_Dets',
    'MT_Startup_Investors_Dets',
    'MT_Startup_FAQs_Dets',
    'MT_Startup_Pressinfo',
    'MT_Startup_Analystics_TRinfo',    
    'Campaign_Banner',
    'Campaign_Press',
    'Campaign_Investors',
    'Campaign_FAQs',
    'E_sign',
    'Campaign_Analystics'
]


MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'corsheaders.middleware.CorsMiddleware', #newly added
    'django.middleware.common.CommonMiddleware',  #newly added
]

ROOT_URLCONF = 'Backend.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [
             os.path.join(REAL_BASE_DIR,'Myntinvest','build')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'Backend.wsgi.application'
 

# Database
# https://docs.djangoproject.com/en/4.0/ref/settings/#databases

 

DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.mysql',
         'NAME': 'livewellmyinvestdb',
         'USER':'admin',
         'PASSWORD':'8Ny9auml',
         'HOST':'livewellmyinvestdb.c6zyqmtnvcnm.ap-south-1.rds.amazonaws.com',
         'PORT':'3306'
     } 
 }


# Password validation
# https://docs.djangoproject.com/en/4.0/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/4.0/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.0/howto/static-files/

STATIC_URL = '/static/'

# Default primary key field type
# https://docs.djangoproject.com/en/4.0/ref/settings/#default-auto-field
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
MEDIA_URL = '/media/'




#EMAIL CONFIGURATION
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_USE_TLS = True
EMAIL_PORT = 587
EMAIL_HOST_USER = 'thulasy2022@gmail.com'
EMAIL_HOST_PASSWORD = 'shvftbdfcpcjwepx'
EMAIL_USE_SSL = False
#############

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
CORS_ORIGIN_ALLOW_ALL = True
CORS_ALLOWED_ORIGINS = [
    "https://example.com",
    "https://sub.example.com",
    "https://master.d3f0xcliax94zn.amplifyapp.com",
    "http://127.0.0.1:3000",
]
STATICFILES_DIRS = [
     os.path.join(REAL_BASE_DIR,'Myntinvest','build','static'),
]
DEBUG = True

Backend/settings.py
"""
Django settings for Backend project.

Generated by 'django-admin startproject' using Django 4.0.6.

For more information on this file, see
https://docs.djangoproject.com/en/4.0/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.0/ref/settings/
"""

from pathlib import Path
import os

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__))) 
REAL_BASE_DIR = Path(__file__).resolve().parent.parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.0/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-oe^*sm5lluuhb)gm8pcu$xa$!kdwu48j+#%1h86gt=d8()$k!k'
 

ALLOWED_HOSTS = []
# ALLOWED_HOSTS = ['3.110.136.18','ec2-3-110-136-186.ap-south-1.compute.amazonaws.com','127.0.0.1']

# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles', 
    'rest_framework',#newly added https://www.bezkoder.com/django-react-axios-rest-framework/
    'corsheaders',#newly added https://www.bezkoder.com/react-crud-web-api/
    'MT_User', #newly added 
    'MT_Investor_Verify_Standards', 
    'MT_Investor_Verifiy_Nationality', 
    'MT_Investor_Kyc',
    'MT_Investor_Payment',
    'MT_Investor_Choose_Sector',
    'MT_Raise',
    'MT_Startup_Agreements',
    'MT_Startup_Companyinfo',
    'MT_Startup_Teaminfo',
    'MT_Startup_Pitchinfo',
    'MT_Startup_Pitch_Probleminfo',
    'MT_Startup_Pitch_Solution',
    'MT_Startup_Pitch_Productinfo',
    'MT_Startup_Pitch_Tractioninfo',
    'MT_Startup_Pitch_Businfo',
    'MT_Startup_Pitch_Competitioninfo',
    'MT_Startup_Pitch_Customerinfo',
    'MT_Startup_Pitch_Usagefundsinfo',
    'MT_Startup_Pitch_Visioninfo',
    'MT_Startup_Pitch_PotentialReturnsinfo',
    'MT_Startup_Banner_Dets',
    'MT_Startup_Investors_Dets',
    'MT_Startup_FAQs_Dets',
    'MT_Startup_Pressinfo',
    'MT_Startup_Analystics_TRinfo',    
    'Campaign_Banner',
    'Campaign_Press',
    'Campaign_Investors',
    'Campaign_FAQs',
    'E_sign',
    'Campaign_Analystics'
]


MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'corsheaders.middleware.CorsMiddleware', #newly added
    'django.middleware.common.CommonMiddleware',  #newly added
]

ROOT_URLCONF = 'Backend.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [
             os.path.join(REAL_BASE_DIR,'Myntinvest','build')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'Backend.wsgi.application'
 

# Database
# https://docs.djangoproject.com/en/4.0/ref/settings/#databases
# Remote DB is connected
DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.mysql',
         'NAME': 'livewellmyinvestdb',
         'USER':'admin',
         'PASSWORD':'8Ny9auml',
         'HOST':'livewellmyinvestdb.c6zyqmtnvcnm.ap-south-1.rds.amazonaws.com',
         'PORT':'3306'
     } 
 }



# Password validation
# https://docs.djangoproject.com/en/4.0/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/4.0/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.0/howto/static-files/

STATIC_URL = '/static/'

# Default primary key field type
# https://docs.djangoproject.com/en/4.0/ref/settings/#default-auto-field
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
MEDIA_URL = '/media/'




#EMAIL CONFIGURATION
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_USE_TLS = True
EMAIL_PORT = 587
EMAIL_HOST_USER = 'thulasy2022@gmail.com'
EMAIL_HOST_PASSWORD = 'shvftbdfcpcjwepx'
EMAIL_USE_SSL = False
#############

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
CORS_ORIGIN_ALLOW_ALL = True
CORS_ALLOWED_ORIGINS = [
    "https://example.com",
    "https://sub.example.com",
    "http://localhost:3000",
    "http://127.0.0.1:3000",
]
STATICFILES_DIRS = [
     os.path.join(REAL_BASE_DIR,'Myntinvest','build','static'),
]
DEBUG = True